File: spec/python/test_default_endian_mod.py
import unittest

from default_endian_mod import DefaultEndianMod

class TestDefaultEndianMod(unittest.TestCase):
    def test_default_endian_mod(self):
        r = DefaultEndianMod.from_file("src/fixed_struct.bin")

        self.assertEqual(r.main.one, 0x4b434150)
        self.assertEqual(r.main.nest.two, -52947)
        self.assertEqual(r.main.nest_be.two, 0x5041434b)

File: spec/lua/test_default_endian_mod.lua
local luaunit = require("luaunit")

require("default_endian_mod")

TestDefaultEndianMod = {}

function TestDefaultEndianMod:test_default_endian_mod()
    local r = DefaultEndianMod:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r.main.one, 0x4b434150)
    luaunit.assertEquals(r.main.nest.two, -52947)
    luaunit.assertEquals(r.main.nest_be.two, 0x5041434b)
end
